src/internal/task/util.go

```diff
@@ -3,16 +3,45 @@ package task
 import (
 	"context"
 
-	taskapi "github.com/pachyderm/pachyderm/v2/src/task"
-	"go.uber.org/zap"
-
 	"github.com/gogo/protobuf/jsonpb"
 	"github.com/gogo/protobuf/types"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+
+	taskapi "github.com/pachyderm/pachyderm/v2/src/task"
+
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/log"
-	"golang.org/x/sync/errgroup"
+	"github.com/pachyderm/pachyderm/v2/src/internal/taskchain"
 )
 
+// DoOrdered processes tasks in parallel, but returns objects in order via the provided callback cb.
```

Comment:
nit, objects -> outputs

Reply:
We should probably also have a note about the index in the collect function not being supported.